src/Microsoft.DotNet.Build.Tasks/PackageFiles/publishtest.targets

```diff
@@ -182,32 +182,35 @@
     </Copy>
   </Target>
 
-  <!-- Workaround for VS execution:  This will form the same list and copy the same files as
-       copied via RunTests script so VS can work when the test dir is initially clean.
+  <!-- This forms the same list and copy the same files as copied via RunTests script, when running locally.
+       This achieves two things: 
+	   1) Allows VS to work when the test dir is initially clean, since VS does not know how to invoke the runner script
+	   2) For incremental build of any kind, ensures that if direct dependencies that also are in the output of 
+	      PrereleaseResolveNuGetPackageAssets are updated, they get copied (thanks to smarter copy logic)
```

Comment:
This target shouldn't know anything about PrereleaseResolveNugetPackageAssets, only the target that is building the file list should care about that.